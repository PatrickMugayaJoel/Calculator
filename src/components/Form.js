import React from 'react';

const Form = ({props}) => {
    return (<form onSubmit={props.onSubmit}>
                <label>Number one</label>
                <input
                    type="number"
                    name="num_one"
                    className="input"
                    required
                    onChange={props.onChange}
                    value={props.state.num_one} 
                />
                <label>Number two</label>
                <input
                    type="number"
                    name="num_two"
                    className="input"
                    required
                    onChange={props.onChange}
                    value={props.state.num_two} 
                />
                <label>Operation</label>
                <select
                    type="text"
                    name="operation"
                    className="input"
                    onChange={props.onChange}
                >
                    <option value="add">ADD</option>
                    <option value="sub">SUB</option>
                    <option value="mul">MUL</option>
                    <option value="div">DIV</option>
                </select>
                <input type="submit" className="submit" value="Post"/>
          </form>);
}
export default Form;