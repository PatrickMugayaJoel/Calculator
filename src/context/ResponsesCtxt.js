import React, {useState, createContext} from 'react';

export const ResponseContext = createContext();

export const ResponseProvider = props => {
    const [responces, setResponces] = useState([]);
    return (
        <ResponseContext.Provider value={[responces, setResponces]}>
            {props.children}
        </ResponseContext.Provider>
    );
}
