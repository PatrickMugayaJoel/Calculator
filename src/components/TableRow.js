import React, { useContext } from 'react';
import {ResponseContext} from "../context/ResponsesCtxt";

const TableRow = ({result, id}) => {
    const [responces, setResponces] = useContext(ResponseContext);

    const PopResponse = index => {
        const list = [...responces];
        console.log(responces);
        console.log(index);
        list.splice(index, 1);
        console.log(list);
        setResponces(list);
    }
    return (
        <tr className={result.is_passed === "No" ? "wrong" : "correct"}>
            <td>{result.num_1}</td>
            <td>{result.num_2}</td>
            <td>{result.resp}</td>
            <td>{result.expected}</td>
            <td>{result.is_passed}</td>
            <td>
                <span
                    className="close"
                    onClick={()=>{PopResponse(id)}}
                >x</span>
            </td>
        </tr>
    );
}

export default TableRow;
