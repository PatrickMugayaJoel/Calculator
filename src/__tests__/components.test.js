import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';
import TableRow from '../components/TableRow';
import Form from '../components/Form';
import {ResponseProvider} from "../context/ResponsesCtxt";

const tableProps = {
  result: {
      expected: "2",
      is_passed: "No",
      num_1: "1",
      num_2: "1",
      resp: "2"
    },
    id: 0
}

const formProps = {
    props: {
        onSubmit: jest.fn(),
        onChange: jest.fn(),
        state: {
            num_one: 1,
            num_two: 2
        }
    }
}

describe('TableRow component snapshot', () => {
    let wrapper = shallow(
        <ResponseProvider>
          <TableRow {...tableProps} />
        </ResponseProvider>
      );
  
    it('matches snapshot', () => {
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });
  

describe('Form component snapshot', () => {
  let wrapper = shallow(<Form {...formProps} />);

  it('matches snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  
  it('calls onSubmit prop function when form is submitted', () => {
    const form = wrapper.find('form');
    form.simulate('submit');
    expect(formProps.props.onSubmit).toHaveBeenCalledTimes(1);
  });

});
