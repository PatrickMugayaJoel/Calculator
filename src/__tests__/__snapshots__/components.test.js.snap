// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form component snapshot matches snapshot 1`] = `
<form
  onSubmit={[MockFunction]}
>
  <label>
    Number one
  </label>
  <input
    className="input"
    name="num_one"
    onChange={[MockFunction]}
    placeholder="Enter number one"
    required={true}
    type="number"
    value={1}
  />
  <label>
    Number two
  </label>
  <input
    className="input"
    name="num_two"
    onChange={[MockFunction]}
    placeholder="Enter number two"
    required={true}
    type="number"
    value={2}
  />
  <label>
    Operation
  </label>
  <select
    className="input"
    name="operation"
    onChange={[MockFunction]}
    type="text"
  >
    <option
      value="add"
    >
      ADD
    </option>
    <option
      value="sub"
    >
      SUB
    </option>
    <option
      value="mul"
    >
      MUL
    </option>
    <option
      value="div"
    >
      DIV
    </option>
  </select>
  <input
    className="submit"
    type="submit"
    value="Post"
  />
</form>
`;

exports[`TableRow component snapshot matches snapshot 1`] = `
<ContextProvider
  value={
    Array [
      Array [],
      [Function],
    ]
  }
>
  <TableRow
    id={0}
    result={
      Object {
        "expected": "2",
        "is_passed": "No",
        "num_1": "1",
        "num_2": "1",
        "resp": "2",
      }
    }
  />
</ContextProvider>
`;
